controllers/datadogagent/feature/admissioncontroller/feature_test.go

```diff
@@ -0,0 +1,129 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
```

Comment:
I think it would be good to also add an integration test for this feature in `datadogagent_controller_v2_test.go`.